### PR TITLE
Make subrequests easier useable for plugins

### PR DESCRIPTION
**Pull request by shyim**
### 1. Why is this change necessary?
Moved code from action tag plugin to Enlight_Controller_Front::subRequest

### 2. What does this change do, exactly?
Makes easier subrequesting pages, without copying 50 lines logic from action tag. The action tag is now small and cute :smiley: 

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Library/Enlight/Controller/Front.php [hunk 1]
```diff
@@ -466,4 +466,77 @@ public function getParams()
...
+     * @throws Enlight_Exception
+     * @throws Exception
+     */
+    public function subRequest(array $params)
+    {
+        $dispatcher = clone $this->Dispatcher();
+
+        $modules = $dispatcher->getControllerDirectory();
+        if (empty($modules)) {
+            throw new Exception('Action helper depends on valid front controller instance');
```
Comment (mitelg) [inline engine/Library/Enlight/Controller/Front.php:483]: both exceptions still refer to the action smarty plugin

File: engine/Library/Enlight/Template/Plugins/function.action.php [hunk 2]
```diff
@@ -36,75 +36,11 @@
  */
 function smarty_function_action($params, Enlight_Template_Default $template)
 {
-    /** @var $front Enlight_Controller_Front */
-    $front = Shopware()->Front();
-    $dispatcher = clone $front->Dispatcher();
+    $response = Shopware()->Front()->subRequest($params);
```
Comment (dneustadt) [inline engine/Library/Enlight/Template/Plugins/function.action.php:40]: Maybe add a filter event while you're at it? Can't have enough of those.
`
Shopware()->Events()->filter(
    'Enlight_Template_Plugins_ActionFunction_FilterResponse',
    $response,
    ['params' => $params]
);
`

Comment (hlohaus) [reply engine/Library/Enlight/Template/Plugins/function.action.php:40]: Or a service with an interface.

**Comment by shyim** [comment]
I will rework this Pull Request and open it internal

**Pull request closed without merging**